src/main/java/uk/nhs/hee/tis/trainee/sync/service/PlacementSpecialtySyncService.java

```diff
@@ -82,4 +87,10 @@ public void request(String id) {
     }
   }
 
+  private boolean haveSameSpecialtyIds(Record record, PlacementSpecialty placementSpecialty) {
+    String incomingSpecialtyId = record.getData().get("specialtyId");
+    String storedSpecialtyId = placementSpecialty.getData().get("specialtyId");
+    return incomingSpecialtyId.equals(storedSpecialtyId);
```

Comment:
I'm anticipating someone else suggesting you use `Objects.equals(record.getData().get("specialtyId"), placementSpecialty.getData().get("specialtyId"))`.  Amongst other things it is "null safe" for the `Map.get()`

Reply:
Refactored.